src/tools/rbd/Shell.cc

```diff
@@ -301,8 +301,14 @@ void Shell::get_command_spec(const std::vector<std::string> &arguments,
 
 Shell::Action *Shell::find_action(const CommandSpec &command_spec,
                                   CommandSpec **matching_spec, bool *is_alias) {
-  for (size_t i = 0; i < get_actions().size(); ++i) {
-    Action *action = get_actions()[i];
+  // sort such that all "trash purge schedule ..." actions come before
+  // "trash purge"
+  std::vector<Action *> actions(get_actions());
+  std::sort(actions.begin(), actions.end(), [](auto lhs, auto rhs) {
+    return lhs->command_spec.size() > rhs->command_spec.size();
+  });
```

Comment:
nit: alternatively it seems we could just use `std::multiset` for storing the actions, not to sort them on every `find_action` call.  But taking that `find_action` is called only once(twice) I don't mind.

Reply:
Actions are also sorted alphabetically when printing help, so I opted for keeping the main vector unsorted and sortitng on-demand with the required comparator.